src/lumigo_opentelemetry/instrumentations/grpcio/grpc_instrument_server.py

```diff
@@ -0,0 +1,70 @@
+from typing import List, Any
+
+from opentelemetry.instrumentation.grpc._server import OpenTelemetryServerInterceptor
+from opentelemetry.trace import get_current_span
+from grpc._utilities import RpcMethodHandler
+from grpc._server import _RequestIterator
+
+from .config import PAYLOAD_MAX_SIZE
+
+
+def request_wrapper(request: _RequestIterator) -> None:
+    history: List[str] = []
+    original_next = request._next
+
+    def wrapped_next() -> Any:
+        data_chunk = original_next()
+        if len(str(history)) < PAYLOAD_MAX_SIZE:
+            history.append(str(data_chunk))
+            get_current_span().set_attribute(
+                "rpc.request.payload", ",".join(history)[:PAYLOAD_MAX_SIZE]
+            )
+        return data_chunk
+
+    request._next = wrapped_next
+
+
+class LumigoServerInterceptor(OpenTelemetryServerInterceptor):
+    def _start_span(self, handler_call_details, context, set_status_on_exception=False):  # type: ignore
```

Comment:
ditto, this also looks very familiar...

Reply:
hmm here it will be hard to merge. 
These methods override different parent methods, so they do different things (although the name is similar).